src/main/kotlin/com/atlassian/performance/tools/report/api/judge/Verdict.kt

```diff
@@ -14,20 +14,42 @@ class Verdict internal constructor(
     val reports: List<JUnitReport>,
     val failedActions: List<ActionType<*>>
 ) {
+    private companion object {
+        private val logger = LogManager.getLogger(this::class.java)
+    }
 
+    @Deprecated("Use Builder")
     constructor(
         reports: List<JUnitReport>
     ) : this(
         reports = reports,
         failedActions = emptyList()
     )
 
-    private val logger = LogManager.getLogger(this::class.java)
+    /**
+     * Since 3.12.0
+     */
+    class Builder(private val reports: List<JUnitReport>) {
```

Comment:
Requiring this makes it painful than the old constructor. Perhaps this should be an empty list by default?

Reply:
There was no usages of previous constructor producing it with an empty list. And it wouldn't really make sense to provide it. It indeed requires more boilerplate, but I think we need to have Builder to have API extensible without breaking it